lib/package-manager/index.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+const npm = require('./npm');
+const yarn = require('./yarn');
+
+function install(context, next) {
+  if (context.options.yarn || context.module.useYarn) {
+    yarn.install(context, next);
```

Comment:
rather than binding the npm / yarn argument on the export of `yarn.install`,  `yarn.test`, `npm.install`, and `npm.test` I think we should be way better off just calling install with the appropriate arguments here. If we cache the result of `whichnpm` above we could alternatively call here with
```
install({name: 'yarn', path: yarnPath}, context, next);
```